packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -377,16 +401,17 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
   /// bar is scrolled.
   final double bottomOpacity;
 
-  /// A size whose height is the sum of [kToolbarHeight] and the [bottom] widget's
+  /// A size whose height is the sum of [kToolbarHeight] (or [titleHeight], if given) and the [bottom] widget's
```

Comment:
I think this should just be `A size whose height is the sum of [titleHeight] and the [bottom] widget's preferred height`, assuming we change title height to never be null.